src/planner/parabolic/ParabolicUtil.cpp

```diff
@@ -128,11 +135,22 @@ std::unique_ptr<ParabolicRamp::DynamicPath> convertToDynamicPath(
     const aikido::trajectory::Spline& _inputTrajectory,
     const Eigen::VectorXd& _maxVelocity,
     const Eigen::VectorXd& _maxAcceleration,
+    statespace::ConstStateSpacePtr& _outputStateSpace,
     bool _preserveWaypointVelocity)
 {
-  const auto stateSpace = _inputTrajectory.getStateSpace();
+  auto stateSpace = _inputTrajectory.getStateSpace();
+  auto metaSkeletonStateSpace
+      = std::dynamic_pointer_cast<const MetaSkeletonStateSpace>(stateSpace);
+
   const auto numWaypoints = _inputTrajectory.getNumWaypoints();
 
+  ConstSplinePtr r1Trajectory;
+  if (metaSkeletonStateSpace)
+  {
+    r1Trajectory = toR1JointTrajectory(stateSpace, _inputTrajectory);
```

Comment:
we should do the same in Kunz timer also

Reply:
Should I add a check here to do this only if all joints are 1D and are either SO2 or R1?